fileparser.py

```diff
@@ -1,29 +1,23 @@
-'''
-fileparser.py - contains basic functions to perform file operations
-'''
-
-class fo:
+class fp:
     def __init__(self, fname):
         self.fname = fname
         self.fname_length = len(fname)
-        self.fname_extension = self.get_extension(fname)
-    
-    def get_extension(self, fname):
-        # use this to introduce methods and attributes
-        i = fname.rfind('.')
-        extension = fname[i+1:]
-        return extension
-    
+        self.extension = self._get_extension()
+        
     def convert_to_list(self):
-        lines = []
-        with open(self.fname) as f:
+        fname_lines = []
+        with open(self.fname, 'r') as f:
             for line in f:
-                lines.append(line.rstrip('\n'))
-        return lines
+                fname_lines.append(line.rstrip('\n'))
+        return fname_lines
+
+    def _get_extension(self):
```

Comment:
The underscore at the start of the method is absent from the `get_extension` method definition in the Jupyter lesson. Do we need it here? Technically it should be there but maybe not worth it in the lesson since that would require explaining methods that are meant for internal vs. external use.

Reply:
I actually do briefly touch on internal vs external methods at the very end! After using it to dynamically update the `extension` attribute I'm having them switch it to a private method just so they know to recognize that in the packages they end up using. Adding a header for that section to emphasize it more.